File: LeetCode_ProblemSet/Problem3169.cpp
#include <vector>
#include <algorithm>


class Solution {
    public:
    int countDays(int days, std::vector<std::vector<int>>& meetings) {
        int curStart = 0, curEnd = 0;
        int count = 0;

        std::sort(meetings.begin(), meetings.end());
        for (const auto& meeting : meetings) {
            int start = meeting[0], end = meeting[1];

            if (curStart <= start && start <= curEnd) {
                curEnd = std::max(curEnd, end);
            } else {
                count += start - curEnd - 1;
                curStart = start, curEnd = end;
            }
        }
        count += days - curEnd;

        return count;
    }
};

File: LeetCode_ProblemSet/Rust/src/Problem3169.rs
use std::cmp::max;

impl Solution {
    pub fn count_days(days: i32, meetings: Vec<Vec<i32>>) -> i32 {
        let (mut curStart, mut curEnd) = (0, 0);
        let mut count = 0;

        let mut meetings_copy = meetings;
        meetings_copy.sort();

        for meeting in meetings_copy {
            let (start, end) = (meeting[0], meeting[1]);

            if curStart <= start && start <= curEnd {
                curEnd = max(curEnd, end);
            } else {
                count += start - curEnd - 1;
                (curStart, curEnd) = (start, end);
            }
        }
        count += days - curEnd;

        return count;
    }
}